Source/WebCore/rendering/style/TextSizeAdjustment.h

```diff
@@ -28,18 +28,19 @@ namespace WebCore {
 
 class RenderStyle;
 
-enum TextSizeAdjustmentType { AutoTextSizeAdjustment = -1, NoTextSizeAdjustment = -2 };
-
 class TextSizeAdjustment {
 public:
-    constexpr TextSizeAdjustment() : m_value(AutoTextSizeAdjustment) { }
+    static constexpr float Auto = -1;
+    static constexpr float None = -2;
+
+    constexpr TextSizeAdjustment() : m_value(TextSizeAdjustment::Auto) { }
     constexpr TextSizeAdjustment(float value) : m_value(value) { }
 
-    constexpr float percentage() const { return m_value; }
-    constexpr float multiplier() const { return m_value / 100; }
+    constexpr float percentage() const { ASSERT(m_value >= 0); return m_value; }
+    constexpr float multiplier() const { ASSERT(m_value >= 0); return m_value / 100; }
```

Comment:
I think this needs to be `ASSERT_IN_CONTEXPR_CONTEXT`.

Reply:
I did a Debug build, but the build succeeded.

Does `ASSERT_IN_CONSTEXPR_CONTEXT()` exist today?  Did it literally just get committed this afternoon?  I can't seem to find it in WTF.  Oh, it's `ASSERT_UNDER_CONSTEXPR_CONTEXT()`.  Sorry--grep commands failed me.

Reply:
I fixed my wrong suggestion.